spki

```diff
@@ -173,32 +189,32 @@ init () {
 	echoc '*Consider generating it on an air-gapped system and using `shred` to remove it.' red
 	echoc '********************************************************************************' red
 	echo
-	if [[ -n "$ROOT_CRL_DP" || -n "$INTRMDT_CRL_DP" ]]; then
+	if [[ -n $ROOT_CRL_DP || -n $INTRMDT_CRL_DP ]]; then
 		echoc 'The following CRLs were created:' green
-		if [[ -n "$ROOT_CRL_DP" ]]; then
+		if [[ -n $ROOT_CRL_DP ]]; then
 			echoc "Root CA: $ROOT_CRL" | indent
 		fi
-		if [[ -n "INTRMDT_CRL_DP" ]]; then
+		if [[ -n $INTRMDT_CRL_DP ]]; then
 			echoc "Intermediate CA: $INTRMDT_CRL" | indent
 		fi
 		echo
 		echoc 'CRLs must be regenerated at regular intervals (Intermediate: 30 days, Root: 4 years). Use `spki generate-crl`' yellow | indent
 		echo
 	fi
-	if [[ -n "$ROOT_OCSP" || -n "$INTRMDT_OCSP" ]]; then
+	if [[ -n $ROOT_OCSP || -n $INTRMDT_OCSP ]]; then
 		echoc 'The following OCSP certificates were created:' green
-		if [[ -n "$ROOT_OCSP" ]]; then
+		if [[ -n $ROOT_OCSP ]]; then
 			echoc "Root CA: $ROOT_OCSP_CERT" | indent
 		fi
-		if [[ -n "INTRMDT_OCSP" ]]; then
+		if [[ -n $INTRMDT_OCSP ]]; then
```

Comment:
thanks for catching the missing `$`. Since these are user supplied vars we should quote them all

Reply:
Actually you don't have to quote them. 
You used `[[` syntax that prevents variables from being splitted. 
Here is little example:

```
 2019-12-04 10:13:17 ⌚  flakm-XPS-15-9570 in ~
○ → A="a b"; if [[ $A == "a b" ]]; then echo "this works"; fi
this works

 2019-12-04 10:13:24 ⌚  flakm-XPS-15-9570 in ~
○ → A="a b"; if [ $A == "a b" ]; then echo "this works"; fi
bash: [: too many arguments

 2019-12-04 10:13:29 ⌚  flakm-XPS-15-9570 in ~
○ → A="a b"; if [ "$A" == "a b" ]; then echo "this works"; fi
this works
```

I know i was suprised too ;) Heres the explenation: https://stackoverflow.com/questions/3427872/whats-the-difference-between-and-in-bash

`[[` syntax is not POSIX compliant so spki has a strong bash dependency but it isnt probably the biggest deal ;P

Reply:
And as for catching them you might wanna check out shell check. It is trully glorious tool for messy developers as me ;) https://www.shellcheck.net/ it has planty of installation options. One could even include it in CI pipeline. It was one of the warnings that got missing `$`

Reply:
ci pipline with automated shellcheck and your test script could be super helpfull for future improvements. If you want I can help you set it up